### PR TITLE
feat: Qdrant support for the VectorDB interface

### PR DESCRIPTION
**Pull request by Anush008**
## Description
Adds a Qdrant - https://qdrant.tech/ based implementation for the `VectorDB` interface.

## Checks
- [x] I've written tests to cover the implemented code.

### CODE REVIEW

**Review by thinkall — COMMENTED** [review]
Thank you very much for the PR, @Anush008 !

Would you like to update the notebook example as well?

File: autogen/agentchat/contrib/vectordb/base.py [hunk 1]
```diff
@@ -207,6 +207,10 @@ def create_vector_db(db_type: str, **kwargs) -> VectorDB:
             from .pgvectordb import PGVectorDB
 
             return PGVectorDB(**kwargs)
+        if db_type.lower() in ["qdrant", "qdrantdb"]:
```
[the remark this answers is not in the document]
Comment (Anush008) [reply autogen/agentchat/contrib/vectordb/base.py:210]: Hey @thinkall. I got on with other parts at work. I missed this.
Added in [`5ced18d` (#3035)](https://github.com/microsoft/autogen/pull/3035/commits/5ced18d5e8e3235a36c91f8358c62244d29a9211)

File: autogen/agentchat/contrib/vectordb/qdrant.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+        Initialize the vector database.
+
+        Args:
+            client: qdrant_client.QdrantClient | An instance of QdrantClient.
+            embedding_function: Callable | The embedding function used to generate the vector representation
+                of the documents. Defaults to FastEmbedEmbeddingFunction.
+            collection_options: dict | The options for creating the collection.
+            kwargs: dict | Additional keyword arguments.
+        """
+        self.client: QdrantClient = client
```
Comment (thinkall) [inline autogen/agentchat/contrib/vectordb/qdrant.py:95]: client is not initiated.

**Review by thinkall — CHANGES_REQUESTED** [review]
Hi @Anush008 , when I tried to use it in a notebook, I saw more issues: `get_docs_by_ids` is not working as expected as well. Could you address the issues and see it work in a notebook? Thanks.

File: autogen/agentchat/contrib/vectordb/qdrant.py [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+            ids: List[ItemID] | A list of document ids. If None, will return all the documents. Default is None.
+            collection_name: str | The name of the collection. Default is None.
+            include: List[str] | The fields to include. Default is True.
+                If None, will include ["metadatas", "documents"], ids will always be included.
+            kwargs: dict | Additional keyword arguments.
+
+        Returns:
+            List[Document] | The results.
+        """
+        results = self.client.retrieve(collection_name, ids=ids, with_payload=include, with_vectors=True)
```
Comment (thinkall) [inline autogen/agentchat/contrib/vectordb/qdrant.py:254]: It doesn't work  when `ids=None`.

**Review by thinkall — APPROVED** [review]

**Pull request merged**